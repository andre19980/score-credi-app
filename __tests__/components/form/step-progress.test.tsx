import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ProgressLine from "@/components/form/step-progress";

describe("ProgressLine Component", () => {
  it("Step Personal Info", () => {
    const { container } = render(<ProgressLine currentStep={0} />);
    expect(container).toMatchSnapshot();
  });

  it("Step Address case", () => {
    const { container } = render(<ProgressLine currentStep={1} />);
    expect(container).toMatchSnapshot();
  });

  it("Step Income", () => {
    const { container } = render(<ProgressLine currentStep={2} />);
    expect(container).toMatchSnapshot();
  });
});
