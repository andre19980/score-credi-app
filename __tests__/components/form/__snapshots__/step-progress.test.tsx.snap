// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressLine Component Step Address case 1`] = `
<div>
  <ol
    class="flex items-center w-full px-4 sm:px-16"
  >
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 6a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0ZM4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.933 17.933 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6.75V15m6-6v8.25m.503 3.498 4.875-2.437c.381-.19.622-.58.622-1.006V4.82c0-.836-.88-1.38-1.628-1.006l-3.869 1.934c-.317.159-.69.159-1.006 0L9.503 3.252a1.125 1.125 0 0 0-1.006 0L3.622 5.689C3.24 5.88 3 6.27 3 6.695V19.18c0 .836.88 1.38 1.628 1.006l3.869-1.934c.317-.159.69-.159 1.006 0l4.994 2.497c.317.158.69.158 1.006 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-gray-100 dark:bg-gray-700 text-gray-500 dark:text-gray-100 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v12m-3-2.818.879.659c1.171.879 3.07.879 4.242 0 1.172-.879 1.172-2.303 0-3.182C13.536 12.219 12.768 12 12 12c-.725 0-1.45-.22-2.003-.659-1.106-.879-1.106-2.303 0-3.182s2.9-.879 4.006 0l.415.33M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
  </ol>
</div>
`;

exports[`ProgressLine Component Step Income 1`] = `
<div>
  <ol
    class="flex items-center w-full px-4 sm:px-16"
  >
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 6a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0ZM4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.933 17.933 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6.75V15m6-6v8.25m.503 3.498 4.875-2.437c.381-.19.622-.58.622-1.006V4.82c0-.836-.88-1.38-1.628-1.006l-3.869 1.934c-.317.159-.69.159-1.006 0L9.503 3.252a1.125 1.125 0 0 0-1.006 0L3.622 5.689C3.24 5.88 3 6.27 3 6.695V19.18c0 .836.88 1.38 1.628 1.006l3.869-1.934c.317-.159.69-.159 1.006 0l4.994 2.497c.317.158.69.158 1.006 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v12m-3-2.818.879.659c1.171.879 3.07.879 4.242 0 1.172-.879 1.172-2.303 0-3.182C13.536 12.219 12.768 12 12 12c-.725 0-1.45-.22-2.003-.659-1.106-.879-1.106-2.303 0-3.182s2.9-.879 4.006 0l.415.33M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
  </ol>
</div>
`;

exports[`ProgressLine Component Step Personal Info 1`] = `
<div>
  <ol
    class="flex items-center w-full px-4 sm:px-16"
  >
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-blue-100 dark:bg-blue-800 text-blue-600 dark:text-blue-500 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 6a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0ZM4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.933 17.933 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-gray-100 dark:bg-gray-700 text-gray-500 dark:text-gray-100 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6.75V15m6-6v8.25m.503 3.498 4.875-2.437c.381-.19.622-.58.622-1.006V4.82c0-.836-.88-1.38-1.628-1.006l-3.869 1.934c-.317.159-.69.159-1.006 0L9.503 3.252a1.125 1.125 0 0 0-1.006 0L3.622 5.689C3.24 5.88 3 6.27 3 6.695V19.18c0 .836.88 1.38 1.628 1.006l3.869-1.934c.317-.159.69-.159 1.006 0l4.994 2.497c.317.158.69.158 1.006 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
    <li
      class="before:content-[''] before:w-full before:h-1 before:border-b before:border-4 before:inline-block first:before:content-none last:after:content-none flex w-full items-center after:content-[''] after:w-full after:h-1 after:border-b after:border-4 after:inline-block"
    >
      <span
        class="bg-gray-100 dark:bg-gray-700 text-gray-500 dark:text-gray-100 flex items-center justify-center w-10 h-10 rounded-full lg:h-12 lg:w-12 shrink-0"
      >
        <svg
          aria-hidden="true"
          class="size-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v12m-3-2.818.879.659c1.171.879 3.07.879 4.242 0 1.172-.879 1.172-2.303 0-3.182C13.536 12.219 12.768 12 12 12c-.725 0-1.45-.22-2.003-.659-1.106-.879-1.106-2.303 0-3.182s2.9-.879 4.006 0l.415.33M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </li>
  </ol>
</div>
`;
